File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms2/Mat4UniformValue.java
package org.saar.lwjgl.opengl.shaders.uniforms2;

import org.joml.Matrix4f;
import org.joml.Matrix4fc;
import org.saar.maths.utils.Matrix4;

public class Mat4UniformValue extends Mat4Uniform implements UniformValue<Matrix4fc> {

    private final String name;

    private final Matrix4f value = Matrix4.create();

    public Mat4UniformValue(String name) {
        this.name = name;
    }

    @Override
    public final Matrix4fc getUniformValue() {
        return getValue();
    }

    @Override
    public final String getName() {
        return this.name;
    }

    @Override
    public final Matrix4f getValue() {
        return this.value;
    }

    @Override
    public final void setValue(Matrix4fc value) {
        getValue().set(value);
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Mat4UniformValue.kt
package org.saar.lwjgl.opengl.shaders.uniforms

import org.joml.Matrix4f
import org.saar.maths.JomlDelegates
import org.saar.maths.utils.Matrix4

class Mat4UniformValue(
    override val name: String,
    initialValue: Matrix4f = Matrix4.create(),
    override val transpose: Boolean = false
) : Mat4Uniform() {
    override var value: Matrix4f by JomlDelegates.CachedMatrix4f(Matrix4.of(initialValue))
}